### PR TITLE
[Merged by Bors] - chore: solve `not necessary anymore` porting notes

### PR DESCRIPTION
**Pull request by pitmonticone**
Solves porting notes claiming
- "not necessary anymore" 
- "not used anymore"

by simply deleting them.

### CODE REVIEW

File: Mathlib/Data/SetLike/Basic.lean [hunk 1]
```diff
@@ -186,9 +186,6 @@ theorem coe_eq_coe {x y : p} : (x : B) = y ↔ x = y :=
   Subtype.ext_iff_val.symm
 #align set_like.coe_eq_coe SetLike.coe_eq_coe
 
--- porting note: this is not necessary anymore due to the way coercions work
-#noalign set_like.coe_mk
```
Comment (jcommelin) [inline Mathlib/Data/SetLike/Basic.lean:190]: I think it's good to keep the `#noalign`, until we drop all `#align` statements.

**Review by jcommelin — COMMENTED** [review]
bors d+

File: Mathlib/Data/SetLike/Basic.lean (hunk 1 above)
Comment (pitmonticone) [reply Mathlib/Data/SetLike/Basic.lean:190]: @jcommelin Done.

**Pull request closed without merging**